### ISSUE TITLE
[4.0] Install email field seen as password field

### ISSUE THREAD

**Issue by Webdongle** [joomla/joomla-cms#18793]
### Steps to reproduce the issue
Install 4.0.0 dev
Click in the email field (when you have saved logins)

### Expected result
List of email addresses display (or none)

### Actual result
List of users/pass are shown.

**Comment by infograf768** [comment]
Confirmed

**Comment by dgrammatiko** [comment]
> List of email addresses display

This comes from your browser and has nothing to do with Joomla!

Please close this, it's not an issue, it's the default browser behaviour

**Comment by ghost** [comment]
closed as stated above.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18793">issues.joomla.org/tracker/joomla-cms/18793</a>.</sub>

**Comment by Webdongle** [comment]
@franz-wohlkoenig and @dgt41 
The browser picks it up from the 'field type'.  The form should have a _input type="password"_ tag<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18793">issues.joomla.org/tracker/joomla-cms/18793</a>.</sub>

**Comment by ghost** [comment]
reopened for above Reason.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18793">issues.joomla.org/tracker/joomla-cms/18793</a>.</sub>

**Comment by C-Lodder** [comment]
@Webdongle which field are you referring to? Can you provide a screenshot please?

**Comment by dgrammatiko** [comment]
@Webdongle if you're talking about the installation (which I assumed) then the code is correct:
<img width="688" alt="screen shot 2017-11-22 at 20 14 55" src="https://user-images.githubusercontent.com/3889375/33145773-00ca3246-cfc2-11e7-85f0-90933289f215.png">

**Comment by Webdongle** [comment]
Yes it is but for some reason the browser (even though the type email is rendered correctly) is not picking it up correctly.  This problem did not manifest before because the order of the fields was email, user, password.  For some reason the browser is not recognising the type with  _<?php echo $this->form->getInput('admin_email'); ?>_  Placing _autocomplete="off"_ in the form tag of the default.php (it is not being recognised in the xml file) prevents it or perhaps put the order the same as installs of 3.4 email, user, password. <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18793">issues.joomla.org/tracker/joomla-cms/18793</a>.</sub>

**Comment by Webdongle** [comment]
It happens in Firefox, Chrome and Edge.  With Edge the _autocomplete="off"_ in the password field of the xml file appears to be recognised.  But in Firefox and Chrome that value is ignored.
![install 01](https://user-images.githubusercontent.com/1115202/33148130-8560ab98-cfc2-11e7-96e9-21a4518b3317.JPG)
![install 02](https://user-images.githubusercontent.com/1115202/33148155-94c447b6-cfc2-11e7-825b-59a9541c6fb7.JPG)

**Comment by dgrammatiko** [comment]
Ok, something is wrong here. If we **cannot** render proper HTML5 forms we're failing miserably. Anyways this is valid and needs to be properly fixed, eg not only for this form but for all forms rendered through JForm

**Comment by Webdongle** [comment]
@dgt41 
The source output looks correct and _type="email"_ shows but the browser just doesn't appear to honour it.  Methinks a few tests need to be done with plain text forms and see if if it a browser fault that somehow overrides the _type_ of a field if it immediately follows a text field ?

**Comment by infograf768** [comment]
This concerns all occurences of the type="email" field in J!, not only during installation.
It is working fine in 3.8.x

**Comment by brianteeman** [comment]
I cannot confirm that. I just checked in the user manager and the email is displayed
Tested in firefox and chrome for windows

**Comment by infograf768** [comment]
My tests here are on Macintosh Firefox and it works perfectly on 3.8.x

**Comment by brianteeman** [comment]
Sorry I misunderstood the report.

I can confirm the behaviour but this looks to me like a firefox browser behaviour (although it doesnt happen in j3

**Comment by Webdongle** [comment]
I suspect the problem existed in J3 but didn't show up because the email field was presented before the user field.
If in J4 \installation\tmpl\setup\default.php is changed from
`					<?php echo $this->form->getLabel('admin_user'); ?>
					<?php echo $this->form->getInput('admin_user'); ?>
				</div>
				<div class="form-group">
					<?php echo $this->form->getLabel('admin_email'); ?>
					<?php echo $this->form->getInput('admin_email'); ?>`
to
`				<div class="form-group">
					<?php echo $this->form->getLabel('admin_email'); ?>
					<?php echo $this->form->getInput('admin_email'); ?>
				</div>
				<div class="form-group">
					<?php echo $this->form->getLabel('admin_user'); ?>
					<?php echo $this->form->getInput('admin_user'); ?>
				</div>`
then the problem disappears

**Comment by Webdongle** [comment]
This is messy but solves it
`				<div class="form-group">
					<?php echo $this->form->getLabel('admin_email'); ?>
					<?php echo $this->form->getInput('admin_email'); ?>
				</div>
					<div style="display: none;">
						<input type="text" id="FakeInput" />
					</div>`
Add the field into the xml file then use echo in the default.php

As for autocomplete="off" ... browsers now only recognise it when in the `<form>` tag not a `<input>` tag !!!

Edited for clarity thanks @dgt41

**Comment by dgrammatiko** [comment]
> As for autocomplete="off" ... browsers now only recognise it when in the `form` tag not a tag !!!

what do you mean?

PS never mind, GitHub didn't display the `<for>` part

**Comment by infograf768** [comment]
@Webdongle Can you make a PR?

**Comment by Webdongle** [comment]
What for ...
The altering the order fix
or
The adding of a hidden input to catch the error ?

**Comment by brianteeman** [comment]
Not at my PC right now. But isn't this a Firefox feature offering to autofill the saved email address for this saved username

**Comment by Webdongle** [comment]
@brianteeman 
No it's Firefox offering to fill the email field with a password
![install 01](https://user-images.githubusercontent.com/1115202/33216757-2fc3c050-d12d-11e7-88ca-80b585fd3123.JPG)
I suggested two fixes
1. Put the email field above the user field https://github.com/joomla/joomla-cms/issues/18793#issuecomment-346643200
2.  Add a dummy field after the email field https://github.com/joomla/joomla-cms/issues/18793#issuecomment-346659685 (define in the xml and call in the default.php
Both allow the browser to present mail addresses (instead of password) suggestions for the email field.
@infograf768 asked if I could provide a PR and I asked for which solution.

Also browsers are ignoring _autofill=off_ unless it's in the `<form>` tag. At the moment the password field has  _autofill=off_ (defined in the xml file) but it's being ignored ... so it is redundant there.

**Comment by brianteeman** [comment]
I just checked and firefox is offering to fill the email field with the email address associated with that saved username - see movie below. This is NOT a bug - this is the expected behaviour with firefox. We should not be doing stuff to break the expected behaviour of a web browser.

![password](https://user-images.githubusercontent.com/1296369/33219414-06c05c56-d13a-11e7-9052-879d1dc5cbe8.gif)

autofill=off is something that browsers constantly change between overriding and not

**Comment by Webdongle** [comment]
Not in the user screen.  During install 
![gif](https://user-images.githubusercontent.com/1115202/33223488-cd24090c-d157-11e7-81a5-cb54eab24cbf.gif)


Both the fixes I suggested allow FF to show email addresses in the email field

**Comment by infograf768** [comment]
@Webdongle I prefer not changing the order, therefore using the dummy field.
I added it in setup/default.php
```
[...]
				<div class="form-group">
					<?php echo $this->form->getLabel('admin_email'); ?>
					<?php echo $this->form->getInput('admin_email'); ?>
				</div>
				<div style="display: none;">
					<input type="text" id="dummy"/>
				</div>
				<div class="form-group">
					<?php echo $this->form->getLabel('admin_password'); ?>
					<?php echo $this->form->getInput('admin_password'); ?>
				</div>
[...]
```
and in the forms/setup.xml
```
[...]
		<field
				name="admin_email"
				type="email"
				id="admin_email"
				class="form-control"
				label="INSTL_ADMIN_EMAIL_DESC"
				validate="email"
				required="true"
		/>

		<field
				type="text"
				id="dummy"
		/>

		<field
				name="admin_user"
[...]
```
and it works fine. 👍 
![email](https://user-images.githubusercontent.com/869724/33229520-d1c0bc5c-d1d0-11e7-9147-1b221726fd35.png)

**Comment by Webdongle** [comment]
@infograf768 
OK I will give it a go creating a PR unless someone else does it first
`<div class="form-group">
					<?php echo $this->form->getInput('dummy'); ?>
				</div>`
Might be tidier ?

**Comment by Webdongle** [comment]
The order of the fields in the xml file were different to the order in the default.php ... so I also arranged them in the correct order in the xml file for consistency.

**Comment by ghost** [comment]
closed as having Pull Request #18863
